src/hotspot/os/posix/os_posix.cpp

```diff
@@ -146,6 +149,95 @@ void os::check_dump_limit(char* buffer, size_t bufferSize) {
   VMError::record_coredump_status(buffer, success);
 }
 
+bool os::committed_in_range(address start, size_t size, address& committed_start, size_t& committed_size) {
+
+#ifdef _AIX
+  committed_start = start;
+  committed_size = size;
+  return true;
+#else
+
+  int mincore_return_value;
+  const size_t stripe = 1024;  // query this many pages each time
+  mincore_vec_t* vec = (mincore_vec_t*) malloc(stripe + 1, mtInternal);
```

Comment:
This is only 1KB. I would allocate this on-stack. Lets avoid doing malloc for every invocation of this function. Also saves you the work of manually freeing each time.